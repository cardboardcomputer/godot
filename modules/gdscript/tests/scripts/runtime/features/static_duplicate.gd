const PreloadClass = preload("static_duplicate_preload.notest.gd")
const PreloadClassAlias = PreloadClass

func test():
	var dup_preload_one = PreloadClass.duplicate()
	print(dup_preload_one == Vector2.ONE)

	var dup_preload_two = (PreloadClass as GDScript).duplicate()
	print(dup_preload_two is GDScript)

	var dup_preload_alias_one = PreloadClassAlias.duplicate()
	print(dup_preload_alias_one == Vector2.ONE)

	var dup_preload_alias_two = (PreloadClassAlias as GDScript).duplicate()
	print(dup_preload_alias_two is GDScript)

	var PreloadClassAsGDScript = PreloadClass as GDScript
	var dup_preload_class_as_gdscript_one = PreloadClassAsGDScript.duplicate()
	print(dup_preload_class_as_gdscript_one is GDScript)
