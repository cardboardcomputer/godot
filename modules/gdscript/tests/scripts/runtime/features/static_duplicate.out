GDTEST_OK
preload duplicate
true
true
preload duplicate
true
true
preload duplicate
false
